Packages/KeyAppKit/Sources/KeyAppBusiness/Solana/Socket/RealtimeSolanaAccountService.swift

```diff
@@ -0,0 +1,321 @@
+            let splAccountChangeRequest = try JSONSerialization.data(withJSONObject: splAccountChange)
+
+            print("RealtimeSolanaAccountService", String(data: nativeAccountChangeRequest, encoding: .utf8))
+            print("RealtimeSolanaAccountService", String(data: splAccountChangeRequest, encoding: .utf8))
+
+            ws.send(nativeAccountChangeRequest.bytes)
+            ws.send(splAccountChangeRequest.bytes)
+
+            return nil
+        } catch {
+            errorObserver.handleError(error)
+            return error
+        }
+    }
+
+    /// Current running task for fetching latest account state
+    var getLatestAccountsStateTask: Task<Void, Error>?
+
+    /// Get all latests account state for given owner.
+    func getLatestAccountsState() {
+        getLatestAccountsStateTask?.cancel()
+        getLatestAccountsStateTask = Task {
+            do {
+                // Updating native account balance and get spl tokens
+                // TODO: Need to optimize
+                let (balance, splAccounts) = try await(
+                    apiClient.getBalance(account: owner, commitment: "confirmed"),
+                    apiClient.getTokenWallets(account: owner, tokensRepository: tokensService, commitment: "confirmed")
+                )
+
+                if Task.isCancelled { return }
+
+                let solanaAccount = SolanaAccount(
+                    data: Wallet.nativeSolana(
+                        pubkey: owner,
+                        lamport: balance
+                    )
+                )
+
+                let accounts = [solanaAccount] + splAccounts.map { SolanaAccount(data: $0, price: nil) }
+
+                for account in accounts {
+                    accountsSubject.send(account)
+                }
+            } catch {
+                errorObserver.handleError(error)
+            }
+        }
+    }
+
+    func receiveNotification(notification: SolanaNotification<SolanaProgramChange>) {
+        let value = notification.result.value
+
+        let account = value.account
+        let dataStr = value.account.data.first
+
+        if let dataStr, !dataStr.isEmpty {
+            // SPL Token Account Data
+            Task {
+                do {
+                    guard let pubKey = value.pubkey else { return }
+
+                    // Convert to Data
+                    guard let data = Data(base64Encoded: dataStr) else {
+                        return
+                    }
+
+                    // Parse
+                    var reader = BinaryReader(bytes: data.bytes)
+                    let tokenAccountData = try AccountInfo(from: &reader)
+
+                    // Bad code because of O(n)
+                    let tokens = try await tokensService.getTokensList(useCache: true)
+                    let token = tokens.first { $0.address == tokenAccountData.mint.base58EncodedString }
+
+                    // TODO: Add case when token info is invalid
```

Comment:
make a task for it?

Reply:
The method `` apiClient.getTokenWallets(account: owner, tokensRepository: tokensService, commitment: "confirmed")`` tries to parse an unknown token. The same algorithm should be apply to socket flow or using unknown token by default for both flows.

Reply:
i see. could I create a task for it?